contracts/core/fee-calculators/PerformanceFeeCalculator.sol

```diff
@@ -0,0 +1,377 @@
+     * or if a highWatermarkResetFrequency amount of time has passed since last profit fee collection.
+     */
+    function updateFeeState(
+        uint256 _streamingFee,
+        uint256 _profitFee
+    )
+        internal
+    {
+        // Set streaming fee timestamp
+        feeState[msg.sender].lastStreamingFeeTimestamp = block.timestamp;
+
+        uint256 rebalancingSetValue = SetUSDValuation.calculateRebalancingSetValue(msg.sender, oracleWhiteList);
+        uint256 postStreamingValue = rebalancingSetValue.mul(ONE_HUNDRED_PERCENT.sub(_streamingFee)).deScale();
+
+        // If profit fee then set new high watermark and profit fee timestamp
+        if (_profitFee > 0) {
+            feeState[msg.sender].lastProfitFeeTimestamp = block.timestamp;
+            feeState[msg.sender].highWatermark = postStreamingValue;
+        } else if (timeSinceLastProfitFee(msg.sender) >= highWatermarkResetFrequency(msg.sender)) {
+            // If no profit fee and last profit fee was more than highWatermarkResetFrequency seconds ago then reset
+            // high watermark
+            feeState[msg.sender].highWatermark = postStreamingValue;
+            feeState[msg.sender].lastProfitFeeTimestamp = block.timestamp;            
+        }
+    }
+
+    /**
+     * Parses passed in fee parameters from bytestring.
+     *
+     * 
+     * | CallData                     | Location                      |
+     * |------------------------------|-------------------------------|
+     * | profitFeeFrequency           | 32                            |
+     * | highWatermarkResetFrequency  | 64                            |
+     * | profitFeePercentage          | 96                            |
+     * | streamingFeePercentage       | 128                           |
+     *
+     * @param  _callData            Byte string containing fee parameter data
+     * @return feeParameters        Fee parameters
+     */
+    function parsePerformanceFeeCallData(
+        bytes memory _callData
+    )
+        private
+        pure
+        returns (InitFeeParameters memory)
+    {
+        InitFeeParameters memory parameters;
+
+        assembly {
+            mstore(parameters,           mload(add(_callData, 32)))   // profitFeeFrequency
+            mstore(add(parameters, 32),  mload(add(_callData, 64)))   // highWatermarkResetFrequency
+            mstore(add(parameters, 64),  mload(add(_callData, 96)))   // profitFeePercentage
+            mstore(add(parameters, 96),  mload(add(_callData, 128)))  // streamingFeePercentage
+        }
+
+        return parameters;
+    }
+
+    /*
+     * Validates fee parameters. Ensures that both fees are below the max fee percentages and that they are
+     * multiples of a basis point. Also makes sure highWatermarkResetFrequency is greater than profitFeeFrequency.
+     */
+    function validateFeeParameters(
+        InitFeeParameters memory parameters
+    )
+        private
```

Comment:
Let's make it internal - as maybe we'd want to overwrite it in a future iteration.